universum/github_handler.py

```diff
@@ -63,44 +60,50 @@ def __init__(self, *args, **kwargs):
-            response = requests.get(self.settings.trigger_url, params=param_dict)
-            if self.settings.verbose:
-                self.out.log(f"Triggered {response.url}")
-            response.raise_for_status()
-            self.out.log("Successfully triggered")
-
-        else:
-            self.out.log("Unhandled event, skipping...")
+        try:
+            payload = json.loads(self.payload)
+        except json.decoder.JSONDecodeError as error:
+            text = f"Provided payload value could not been parsed as JSON and returned the following error:\n {error}"
+            raise CriticalCiException(text)
+
+        try:
+            if self.settings.event == "check_suite" and (payload["action"] in ["requested", "rerequested"]):
+                url = payload["repository"]["url"] + "/check-runs"
+                data = {"name": "CI tests", "head_sha": payload["check_suite"]["head_sha"]}
+                headers = {'Authorization': f"token {self.get_token(payload['installation']['id'])}",
+                           'Accept': 'application/vnd.github.antiope-preview+json'}
+                self.out.log(f"Sending request to {url}")
+                if self.settings.verbose:
+                    self.out.log(f"Headers are:\n{headers}\nOther passed params are:\n{data}")
+                response = requests.post(url=url, json=data, headers=headers)
+                response.raise_for_status()
+                self.out.log("Successfully created check run")
+
+            elif self.settings.event == "check_run" and \
+                    (payload["action"] in ["requested", "rerequested", "created"]) and \
+                    (str(payload["check_run"]["app"]["id"]) == str(self.settings.integration_id)):
+                param_dict = {
+                    "GIT_REFSPEC": payload["check_run"]["check_suite"]["head_branch"],
+                    "GIT_CHECKOUT_ID": payload["check_run"]["head_sha"],
+                    "GITHUB_CHECK_ID": payload["check_run"]["id"],
+                    "GIT_REPO": payload["repository"]["clone_url"],
+                    "GITHUB_INSTALLATION_ID": payload["installation"]["id"]
+                }
+                self.out.log(f"Triggering {urllib.parse.urljoin(self.settings.trigger_url, '?...')}")
+                response = requests.get(self.settings.trigger_url, params=param_dict)
+                if self.settings.verbose:
+                    self.out.log(f"Triggered {response.url}")
+                response.raise_for_status()
+                self.out.log("Successfully triggered")
+
+            else:
+                self.out.log("Unhandled event, skipping...")
+        except KeyError as error:
+            raise CriticalCiException(f"Could not find key {error} in provided payload:\n{payload}")
+        except TypeError:
+            raise CriticalCiException("Parsed payload JSON does not correspond to expected format")
+        except (requests.HTTPError, requests.ConnectionError, ValueError) as error:
```

Comment:
Each of these exceptions must be tested

Reply:
replaced with TODO until further implementation